docs: add how to db and tx (#170)
canonical/sqlair: docs/howto/query.md

(query)=
# How to query the database

> ## SQLair-proof your types

Reviewer: ```suggestion
## 1. SQLair-proof your types
```

> The first step in building your query is to decide what you would like to put

Reviewer: SQLair-proof*

> See more: {ref}`types` 

> ## Write your query string using the types

Reviewer: ```suggestion
## 2.Write your query string using the types
```

SQLair queries are regular SQL queries with special input and output expressions

> which allow you to reference the Go types directly from SQL.

Reviewer: ```suggestion
that allow you to reference the Go types directly from SQL.
```

Author: Also changed allow to enable.

> See more: {ref}`types` 

## Write your query string using the types

SQLair queries are regular SQL queries with special input and output expressions

> which allow you to reference the Go types directly from the SQL.

Reviewer: ```suggestion
which allow you to reference the Go types directly from SQL.
```

Author: Mean to say, I felt the numbers were a bit out of place in this article cuz we don't use them anywhere else in the how-tos plus making how to query a 5 step process sounds like quite a lot for a "convenience" library

> See more {ref}`input-expression-syntax`, {ref}`output-expression-syntax`.

> ## Prepare your query string to get a statement

Reviewer: ```suggestion
## 3. Prepare your query string to get a statement
```

}
```

> > See more:

Reviewer: Looks like too much space. Please double-check.

Author: If they're in the same paragraph they are rendered as one block, I think this should be fine

[`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare),
[`sqlair.MustPrepare`](https://pkg.go.dev/github.com/canonical/sqlair#MustPrepare)

> ## Execute the statement on the database

Reviewer: ```suggestion
## 4. Execute the statement on the database
```

}
```

> See more:
[`sqlair.Prepare`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare),
[`sqlair.MustPrepare`](https://pkg.go.dev/github.com/canonical/sqlair#MustPrepare)

## Execute the statement on the database

To execute the statement on a SQLair wrapped `DB` or a `TX`, use the `Query`

> method passing as parameters the `Statement` and all the input arguments

Reviewer: ```suggestion
method, passing as parameters the `Statement` and all the input arguments
```

[`Query.Run`](https://pkg.go.dev/github.com/canonical/sqlair#Query.Run)

> ## (Optional) Get the query outcome

Reviewer: ```suggestion
## 5. (Optional) Get the query outcome
```

[`Query.Run`](https://pkg.go.dev/github.com/canonical/sqlair#Query.Run)

## (Optional) Get the query outcome

> To get the outcome, provide a pointer to a `sqlair.Outcome` as the first

Reviewer: ```suggestion
To get the query outcome, use any of the `Get` methods, providing as a first argument a pointer to a `sqlair.Outcome` object.
```
